app/store.go

```diff
@@ -0,0 +1,146 @@
+package app
+
+import (
+	"fmt"
+	"math"
+	"os"
+	"time"
+
+	"github.com/datagovsg/nomad-parametric-autoscaler/logging"
+	"github.com/jmoiron/sqlx"
+	_ "github.com/lib/pq"
+)
+
+// Store records new state updates to a persistent store
+// table schema : timestamp (TIMESTAMP) | state (text)
+type Store struct {
+	db *sqlx.DB
+}
+
+// CreateTablesSQL contains statement for table if doesnt exist
+const CreateTablesSQL = `
+CREATE TABLE autoscaler (
+    timestamp TIMESTAMP,
+    state text
+);
+`
+const (
+	dbhost = "POSTGRES_HOST"
+	dbport = "POSTGRES_PORT"
+	dbuser = "POSTGRES_USER"
+	dbpass = "POSTGRES_PASSWORD"
+	dbname = "POSTGRES_DB"
+)
+
+func dbConfig() (map[string]string, error) {
+	conf := make(map[string]string)
+	host, ok := os.LookupEnv(dbhost)
+	if !ok {
+		return nil, fmt.Errorf("POSTGRES_HOST environment variable required but not set")
+	}
+	port, ok := os.LookupEnv(dbport)
+	if !ok {
+		return nil, fmt.Errorf("POSTGRES_PORT environment variable required but not set")
+	}
+	user, ok := os.LookupEnv(dbuser)
+	if !ok {
+		return nil, fmt.Errorf("POSTGRES_USER environment variable required but not set")
+	}
+	password, ok := os.LookupEnv(dbpass)
+	if !ok {
+		return nil, fmt.Errorf("POSTGRES_PASSWORD environment variable required but not set")
+	}
+	name, ok := os.LookupEnv(dbname)
+	if !ok {
+		return nil, fmt.Errorf("POSTGRES_DB environment variable required but not set")
+	}
+	conf[dbhost] = host
+	conf[dbport] = port
+	conf[dbuser] = user
+	conf[dbpass] = password
+	conf[dbname] = name
+	return conf, nil
+}
+
+// Initialise creates the connection to sqlx.DB and creates an empty table if none exists
+func (st *Store) Initialise() error {
```

Comment:
This is going to be super pedantic, but I suggest renaming this function to `Init` instead because, `Initialize` vs `Initialise` will always be subjectively debatable, just like `color` vs `colour`.